lib/chef/knife/ec2_server_create.rb

```diff
@@ -824,7 +824,7 @@ def create_server_def
           :groups => config[:security_groups],
           :security_group_ids => locate_config_value(:security_group_ids),
           :flavor_id => locate_config_value(:flavor),
-          :key_name => Chef::Config[:knife][:aws_ssh_key_id],
+          :key_name => config[:ssh_key_name] || Chef::Config[:knife][:aws_ssh_key_id],
```

Comment:
@Aliasgar16 
Ohh Ok. Then,
I think we should change  `:aws_ssh_key_id` option to `:ssh_key_name` (i.e make consistent with others option) but also support `:aws_ssh_key_id` with depreciation warning message, eventually we will remove support for `:aws_ssh_key_id`

@adamedx ^ Please comment on this!